feature/experimental/bgp/ate_tests/bgp_always_compare_med/bgp_always_compare_med_test.go

```diff
@@ -0,0 +1,487 @@
+		}
+		status := gnmi.Get(t, dut, nbrPath.SessionState().State())
+		t.Logf("BGP adjacency for %s: %s", nbr, status)
+		if want := oc.Bgp_Neighbor_SessionState_ESTABLISHED; status != want {
+			t.Errorf("BGP peer %s status got %d, want %d", nbr, status, want)
+		}
+	}
+}
+
+// configureATE configures the interfaces and BGP protocols on an ATE, including
+// advertising some(faked) networks over BGP.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) []*ondatra.Flow {
+	topo := ate.Topology().New()
+
+	port1 := ate.Port(t, "port1")
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+
+	port2 := ate.Port(t, "port2")
+	iDut2 := topo.AddInterface(ateDst1.Name).WithPort(port2)
+	iDut2.IPv4().WithAddress(ateDst1.IPv4CIDR()).WithDefaultGateway(dutDst1.IPv4)
+
+	port3 := ate.Port(t, "port3")
+	iDut3 := topo.AddInterface(ateDst2.Name).WithPort(port3)
+	iDut3.IPv4().WithAddress(ateDst2.IPv4CIDR()).WithDefaultGateway(dutDst2.IPv4)
+
+	// Setup ATE BGP route v4 advertisement.
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dutSrc.IPv4).WithLocalASN(ateAS1).
+		WithTypeInternal()
+
+	bgpDut2 := iDut2.BGP()
+	bgpDut2.AddPeer().WithPeerAddress(dutDst1.IPv4).WithLocalASN(ateAS2).
+		WithTypeExternal()
+
+	bgpDut3 := iDut3.BGP()
+	bgpDut3.AddPeer().WithPeerAddress(dutDst2.IPv4).WithLocalASN(ateAS3).
+		WithTypeExternal()
+
+	bgpNeti1 := iDut2.AddNetwork("bgpNeti1") // Advertise same prefixes from both eBGP Peers.
+	bgpNeti1.IPv4().WithAddress(advertisedRoutesv4CIDR).WithCount(routeCount)
+	bgpNeti1.BGP().WithNextHopAddress(ateDst1.IPv4)
+
+	bgpNeti2 := iDut3.AddNetwork("bgpNeti2") // Advertise same prefixes from both eBGP Peers.
+	bgpNeti2.IPv4().WithAddress(advertisedRoutesv4CIDR).WithCount(routeCount)
+	bgpNeti2.BGP().WithNextHopAddress(ateDst2.IPv4)
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	// ATE Traffic Configuration.
+	ethHeader := ondatra.NewEthernetHeader()
+	// BGP V4 Traffic.
+	ipv4Header := ondatra.NewIPv4Header()
+	ipv4Header.WithSrcAddress(ipv4SrcTraffic).DstAddressRange().
+		WithMin(ipv4DstTrafficStart).WithMax(ipv4DstTrafficEnd).
+		WithCount(routeCount)
+	flowipv41 := ate.Traffic().NewFlow(flow1).
+		WithSrcEndpoints(iDut1).
+		WithDstEndpoints(iDut2).
+		WithHeaders(ethHeader, ipv4Header).
+		WithFrameSize(512)
+	flowipv42 := ate.Traffic().NewFlow(flow2).
+		WithSrcEndpoints(iDut1).
+		WithDstEndpoints(iDut3).
+		WithHeaders(ethHeader, ipv4Header).
+		WithFrameSize(512)
+	return []*ondatra.Flow{flowipv41, flowipv42}
+}
+
+// verifyTraffic confirms that every traffic flow has the expected amount of loss (0% or 100%
+// depending on wantLoss, +- 2%).
+func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, flowName string, wantLoss bool) {
+	// Compare traffic loss based on wantLoss.
+	lossPct := gnmi.Get(t, ate, gnmi.OC().Flow(flowName).LossPct().State())
+	if !wantLoss {
```

Comment:
better to have bool conditions have positive polarity, i.e.

```
if wantLoss {
	if lossPct < 100-tolerancePct {
		...
} else {
	if lossPct > tolerancePct {
		...
}
```

Reply:
updated as suggested.